influxdata/chronograf
# fix(ui): polish template variable ux

Closes #5381

* Coerces existing template variables into a variables with a dynamic source
* Correctly displays the values template variables with dynamic sources at query run time, not just creation time
* Correctly runs the meta query against the correct source database when creating a template variable.

## Testing Instructions
1. Checkout `16ac349228228c0370fb93535e6125ab548ae2aa` to get on the last version before the new template variable code was merged. `git checkout 16ac349`
1. Run chronograf and build the ui
1. Create a template variable. There should be no source dropdown
1. Checkout this branch. `git checkout bs_fix_source_template_query_error` (you might need to do `git checkout .` to ignore any changes to `yarn.lock`
1. The template variable you created before should have the proper values.
1. Edit the template variable you created before by clicking the gear icon.
1. You should see the dropdown with `Dynamic` listed.
1. Modify that as you wish and save it. It should work as expected.
1. Create a new template variable, it should work as expected and run the meta query.
